service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -44,22 +44,10 @@
      */
     public abstract List<String> uris();
 
-    // hides implementation details
     public static Builder builder() {
-        return ImmutableServiceConfiguration.builder();
+        return new Builder();
     }
 
-    // hides implementation details
-    public interface Builder {
-
-        Builder apiToken(BearerToken apiToken);
-
-        Builder security(SslConfiguration security);
-
-        Builder uris(Iterable<String> uris);
-
-        Builder from(ServiceConfiguration otherConfig);
-
-        ServiceConfiguration build();
+    public static final class Builder extends ImmutableServiceConfiguration.Builder {
```

Comment:
Only case where I think it matters is compile errors it can produce -- see https://github.com/palantir/http-remoting/pull/111/files#diff-bd6f45eabbff99720347cb52c6b81eb1R39.

I don't have a strong feeling either way -- I think it is ideal for consumers to not have to worry about how they use the API (which would argue for the "hide the implementation" approach), but at the same time am also not a fan of all the boilerplate code required for the interface definitions (whole point of Immutables is to try to move away from boilerplate).